File: 11-container-with-most-water/container-with-most-water.cpp
#include<bits/stdc++.h>
class Solution {
public:
    int maxArea(vector<int>& height) {
        int l=0,e=height.size()-1;
        int maxi=0;
        while(l<e){
            int min_h=min(height[l],height[e]);
            int curr_area=min_h*(e-l);

            maxi=max(maxi,curr_area);

            if(height[l]<height[e]) l++;
            else e--;
        }

        return maxi;
    }
};

File: 11-container-with-most-water/container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        l, e = 0, len(height) - 1
        maxi = 0
        
        while l < e:
            min_h = min(height[l], height[e])
            curr_area = min_h * (e - l)
            
            maxi = max(maxi, curr_area)
            
            if height[l] < height[e]:
                l += 1
            else:
                e -= 1
        
        return maxi